packages/reactive-dict/reactive-dict.js

```diff
@@ -164,6 +164,23 @@ _.extend(ReactiveDict.prototype, {
     
   },
 
+  delete: function(key) {
+    var self      = this;
+    var oldKeys   = self.keys;
+    var didRemove = false;
+
+    if (_.has(oldKeys, key)) {
+      changed(self.keyDeps[key]);
```

Comment:
@dgreensp At first glance totally, but I think some things pull away from that:

1. A more ES6-ish API (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/delete)
2. I think it's simpler to keep the two apart and distinct